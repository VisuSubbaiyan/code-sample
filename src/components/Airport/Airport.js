import React from 'react';
import PropTypes from 'prop-types';
import IconButton from '../IconButton';
import TimeDisplay from '../TimeDisplay';

const Airport = props => (
  <div className="segment" >
    <TimeDisplay time={props.departure} />
    <div className="card">
      <IconButton icon="plane-departure" />
      <div className="title">
        <div>
          <span className="highlight">{props.origin.value}</span>
          {' Airport'}
        </div>
        <div className="terminal-details"><span className="highlight">{'Terminal C '}</span><span className="status-busy">{'Busy'}</span></div>
      </div>
    </div>
  </div>
);

Airport.propTypes = {
  origin: PropTypes.object.isRequired,
  departure: PropTypes.number.isRequired
}

export default Airport;
