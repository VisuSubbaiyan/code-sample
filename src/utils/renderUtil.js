import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
