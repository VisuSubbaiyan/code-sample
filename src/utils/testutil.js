import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

