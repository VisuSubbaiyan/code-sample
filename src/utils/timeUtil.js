import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}