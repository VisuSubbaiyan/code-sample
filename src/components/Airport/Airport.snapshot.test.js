import React from 'react';

import { createMountedComponent } from '../../utils/testutil';
import Airport from './Airport';

import { segments } from '../../data.json';

describe('<Airport /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <Airport {...segments[0]} />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});