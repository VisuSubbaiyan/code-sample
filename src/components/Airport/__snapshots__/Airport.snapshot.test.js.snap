// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Airport /> should render correctly according to props 1`] = `
"<div class=\\"segment\\">
  <h3 class=\\"time\\">15: 13</h3>
  <div class=\\"card\\">
    <div><i class=\\"fa fa-plane-departure\\"></i>
    </div>
    <div class=\\"title\\">
      <div><span class=\\"highlight\\">Alte Schönhauser Straße 3, 10119 Berlin</span> Airport</div>
      <div
      class=\\"terminal-details\\"><span class=\\"highlight\\">Terminal C </span><span class=\\"status-busy\\">Busy</span>
    </div>
  </div>
</div>
</div>"
`;
